app/src/main/kotlin/at/bitfire/davdroid/ui/AppSettingsModel.kt

```diff
@@ -107,4 +113,58 @@ class AppSettingsModel @Inject constructor(
 
     val pushEndpoint = preference.unifiedPushEndpointFlow()
 
+    private val _pushDistributor = MutableStateFlow<String?>(null)
+    val pushDistributor get() = _pushDistributor.asStateFlow()
+
+    private val _pushDistributors = MutableStateFlow<List<PushDistributorInfo>?>(null)
+    val pushDistributors get() = _pushDistributors.asStateFlow()
+
+    fun updatePushDistributor(pushDistributor: String) {
+        viewModelScope.launch(Dispatchers.IO) {
+            UnifiedPush.saveDistributor(context, pushDistributor)
+            UnifiedPush.registerApp(context)
+            _pushDistributor.emit(pushDistributor)
+        }
+    }
+
+
+    init {
+        viewModelScope.launch(Dispatchers.IO) {
+            val savedPushDistributor = UnifiedPush.getSavedDistributor(context)
+            _pushDistributor.emit(savedPushDistributor)
+
+            val pushDistributors = UnifiedPush.getDistributors(context)
+
+            val pushDistributorsData = mutableMapOf<String, Pair<String, Drawable>>()
```

Comment:
No need to have this map here, which is a too complicated construct. We can directly go through the available distributors, generate a `PushDistributorInfo` per entry and then add it to the result list.